### PR TITLE
New general progress indication bar

### PR DESCRIPTION
**Pull request by talarian1**
- [x] All [tests](https://github.com/jfrog/jfrog-cli#tests) passed. If this feature is not already covered by the tests, I added new tests.
- [x] This pull request is on the dev branch.
- [x] I used gofmt for formatting the code before submitting the pull request.
-----

### CODE REVIEW

File: utils/progressbar/progressbar.go [hunk 1]
```diff
@@ -137,71 +143,16 @@ func createSpinnerFramesArray() []string {
 	return spinnerFramesArray
```
Comment (RobiNino) [inline utils/progressbar/progressbar.go:143]: The white color in this function should be changed to green to match the change in the reader progress bar

**Review by RobiNino — CHANGES_REQUESTED** [review]
Looks great!

File: utils/progressbar/readerprogressbar.go [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+	Id int
+}
+
+// Wraps an io.Reader for bytes reading tracking
+type proxyReader struct {
+	unit *progressBarUnit
+	io.ReadCloser
+}
+
+// Uses to updated the progress bar progress.
```
Comment (eyalbe4) [inline utils/progressbar/readerprogressbar.go:19]: Uses to updated... --> Used to update...

File: utils/progressbar/readerprogressbar.go [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+func (p *ReaderProgressBar) ActionWithProgress(reader io.Reader) (results io.Reader) {
+	return p.readWithProgress(reader)
+}
+
+// Abort aborts a progress indication. Called on both successful and unsuccessful operations
```
Comment (eyalbe4) [inline utils/progressbar/readerprogressbar.go:24]: indication --> indicator

**Review by eyalbe4 — CHANGES_REQUESTED** [review]

**Pull request merged**